pkg/services/notification/resource_control/notification.go

```diff
@@ -54,44 +58,89 @@ func UpdateNotificationStatus(notificationId, status string) error {
-}
-
 func DescribeNotifications(ctx context.Context, req *pb.DescribeNotificationsRequest) ([]*models.Notification, uint64, error) {
 	req.NotificationId = stringutil.SimplifyStringList(req.NotificationId)
 	req.ContentType = stringutil.SimplifyStringList(req.ContentType)
 	req.Owner = stringutil.SimplifyStringList(req.Owner)
 	req.Status = stringutil.SimplifyStringList(req.Status)
 
-	limit := dbutil.GetLimit(req.Limit)
-	offset := dbutil.GetOffset(req.Offset)
+	offset := pbutil.GetOffsetFromRequest(req)
+	limit := pbutil.GetLimitFromRequest(req)
 
 	var nfs []*models.Notification
 	var count uint64
 
-	if err := dbutil.GetChain(globalcfg.GetInstance().GetDB().Table(models.TableNotification)).
+	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableNotification)).
 		AddQueryOrderDir(req, models.NfColCreateTime).
 		BuildFilterConditions(req, models.TableNotification).
 		Offset(offset).
 		Limit(limit).
 		Find(&nfs).Error; err != nil {
-		logger.Errorf(ctx, "Describe Notifications failed: %+v", err)
+		logger.Errorf(ctx, "Describe notifications failed, [%+v]", err)
 		return nil, 0, err
 	}
 
-	if err := dbutil.GetChain(globalcfg.GetInstance().GetDB().Table(models.TableNotification)).
+	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableNotification)).
 		BuildFilterConditions(req, models.TableNotification).
 		Count(&count).Error; err != nil {
-		logger.Errorf(ctx, "Describe Notifications count failed: %+v", err)
+		logger.Errorf(ctx, "Describe notifications count failed, [%+v]", err)
 		return nil, 0, err
 	}
 
 	return nfs, count, nil
 }
+
+func UpdateNotifications2Pending(ctx context.Context, nfIds []string) ([]string, error) {
+	db := global.GetInstance().GetDB()
+	tx := db.Begin()
+	db.Table(models.TableNotification).Where(models.NfColId+" in (?)", nfIds).Updates(map[string]interface{}{models.NfColStatus: constants.StatusPending, models.NfColStatusTime: time.Now()})
+
+	if err := tx.Error; err != nil {
+		tx.Rollback()
+		logger.Errorf(ctx, "Update notifications status to pending failed, [%+v].", err)
+		return nil, err
+	}
+
+	tx.Commit()
+	return nfIds, nil
+}
+
+func SplitNotificationIntoTasks(notification *models.Notification) ([]*models.Task, error) {
+	addressInfo, err := models.DecodeAddressInfo(notification.AddressInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	var tasks []*models.Task
+	for notifyType, addresses := range *addressInfo {
+		for _, address := range addresses {
+			directive := &models.TaskDirective{
+				Address:      address,
+				NotifyType:   notifyType,
+				ContentType:  notification.ContentType,
+				Title:        notification.Title,
+				Content:      notification.Content,
+				ShortContent: notification.ShortContent,
+				ExpiredDays:  notification.ExpiredDays,
+			}
+			task := models.NewTask(
+				notification.NotificationId,
+				jsonutil.ToString(directive),
+			)
+			logger.Debugf(nil, "Split notification into tasks[%s] successfully. ", task.TaskId)
```

Comment:
If need ctx, pls add ctx in func parameters
func SplitNotificationIntoTasks(ctx context.Context, notification *models.Notification) 
we will put request id in ctx.

Reply:
updated.